src/psyclone/tests/psyir/nodes/reference_test.py

```diff
@@ -149,6 +150,28 @@ def test_reference_is_array():
     assert ref.is_array is True
 
 
+def test_reference_is_character():
+    '''Test that character references are marked correctly.
+    '''
+    reference = Reference(DataSymbol("char", CHARACTER_TYPE))
+    assert reference.is_character()
+
+    reference = Reference(DataSymbol("int", INTEGER_SINGLE_TYPE))
+    assert not reference.is_character()
+
+    reference = Reference(DataSymbol("real", REAL_TYPE))
+    assert not reference.is_character()
+
+    reference = Reference(DataSymbol("unknown", UnresolvedType()))
+    with pytest.raises(Exception) as excinfo:
+        _ = reference.is_character()
+    assert ("is_character could not resolve whether the expression 'unknown'"
+            " operates on characters." in str(excinfo.value))
+    reference = Reference(DataSymbol("unknown", UnresolvedType()))
+    assert not reference.is_character(unknown_as=False)
+    assert reference.is_character(unknown_as=True)
```

Comment:
We have a rule to put whenever possible to test in a matching path to its implementation, so for a method in
`src/psyclone/psyir/nodes/datanode.py`
we expect the test to be located in:
`src/psyclone/tests/psyir/nodes/datanode_test.py`

Which make them easier to find in the future.

Also s/reference/datanode/ in the test name.

Reply:
Moved in 5c17b4a3c23acb823bee6e667f92b5027292e6df.